Macros.h

```diff
@@ -142,4 +151,32 @@ static inline unsigned long long saturatingSub(unsigned long long a, unsigned lo
    return a > b ? a - b : 0;
 }
 
+#ifdef HAVE_ALIGNAS
+#include <stdalign.h>
+
+#define ELF_NOTE_DLOPEN_OWNER "FDO"
+#define ELF_NOTE_DLOPEN_TYPE  UINT32_C(0x407c0c0a)
+
+#define DECLARE_ELF_NOTE_DLOPEN(content) \
+   IGNORE_W11EXTENSIONS_BEGIN                                                                      \
+   __attribute__((used, section(".note.dlopen"))) alignas(sizeof(uint32_t)) static const struct {  \
+      struct {                                                                                     \
+         uint32_t n_namesz, n_descsz, n_type;                                                      \
+      } nhdr;                                                                                      \
+      char name[sizeof(ELF_NOTE_DLOPEN_OWNER)];                                                    \
+      alignas(sizeof(uint32_t)) char dlopen_content[sizeof(content)];                              \
```

Comment:
```suggestion
      char dlopen_content[(sizeof(content)+3)&(~3)];                                               \
```

Although one could argue if this last expansion is still required at the end of the attribute section.